### PR TITLE
added new parameters to RepositoryRequest.cs

### PR DESCRIPTION
**Pull request by Sarmad93**
As per the issue  #1065 opened by @shiftkey, i have added new parameters to RepositoryRequest class.

### CODE REVIEW

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 1]
```diff
@@ -103,4 +121,70 @@ public enum RepositorySort
...
+        OwnerAndOrganizationMember,
+
+        /// <summary>
+        /// Return repositories that user has been added as collaborator and user is a member of organization.
+        /// </summary>
+        [Parameter(Value = "collaborator, organization_member")]
+        CollaboratorAndOrganizationMember,
+
+        /// <summary>
+        /// Returns all repositories where user is owner,collaborator and organization member.
```
Comment (M-Zuber) [inline Octokit/Models/Request/RepositoryRequest.cs]: Should probably be `or organization member`

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 2]
```diff
@@ -37,6 +37,24 @@ public class RepositoryRequest : RequestParameters
         /// </value>
         public SortDirection Direction { get; set; }
 
+        /// <summary>
+        /// Gets or sets the visibility property.
+        /// </summary>
+        /// <value>
+        ///  The public.
+        /// </value>
+
```
Comment (M-Zuber) [inline Octokit/Models/Request/RepositoryRequest.cs]: Extra line

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 3]
```diff
@@ -37,6 +37,24 @@ public class RepositoryRequest : RequestParameters
...
+        public RepositoryVisibility Visibility { get; set; }
+
+        /// <summary>
+        /// Gets or sets the affiliation property.
+        /// </summary>
+        /// <value>
+        ///  The owner.
+        /// </value>
+
```
Comment (M-Zuber) [inline Octokit/Models/Request/RepositoryRequest.cs]: Extra line

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 4]
```diff
@@ -103,4 +121,70 @@ public enum RepositorySort
         [Parameter(Value = "full_name")]
         FullName
     }
+
+
```
Comment (M-Zuber) [inline Octokit/Models/Request/RepositoryRequest.cs]: :lipstick: whitespace

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 5]
```diff
@@ -103,4 +121,70 @@ public enum RepositorySort
...
+        OrganizationMember,
+
+        /// <summary>
+        /// Return repositories that are owned by authenticated user and added to as a collaborator.
+        /// </summary>
+        [Parameter(Value = "owner, collaborator")]
+        OwnerAndCollaborator,
+
+        /// <summary>
+        /// Return repositories that are owned by authenticated user and user is a member of organization.
```
Comment (M-Zuber) [inline Octokit/Models/Request/RepositoryRequest.cs]: `and user` should be `or user`

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 6]
```diff
@@ -103,4 +121,70 @@ public enum RepositorySort
...
+        /// </summary>
+        [Parameter(Value = "owner, organization_member")]
+        OwnerAndOrganizationMember,
+
+        /// <summary>
+        /// Return repositories that user has been added as collaborator and user is a member of organization.
```
Comment (M-Zuber) [inline Octokit/Models/Request/RepositoryRequest.cs]: `and user` should be `or user`

**Comment by M-Zuber** [comment]
There seems to be a lot white space that could be cleaned up. 
If you are using visual studio 2013 or higher, pressing `ctrl + K, d` should take care of the formatting. Otherwise it will just have to take the time that it takes.

**Comment by Sarmad93** [comment]
@M-Zuber thanks for correction! i am fixing this.

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 7]
```diff
@@ -37,6 +37,24 @@ public class RepositoryRequest : RequestParameters
         /// </value>
         public SortDirection Direction { get; set; }
 
+        /// <summary>
+        /// Gets or sets the visibility property.
+        /// </summary>
+        /// <value>
+        ///  The public.
```
Comment (M-Zuber) [inline Octokit/Models/Request/RepositoryRequest.cs]: Should be `The visibility` no?

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 8]
```diff
@@ -37,6 +37,24 @@ public class RepositoryRequest : RequestParameters
...
+        /// </value>
+
+        public RepositoryVisibility Visibility { get; set; }
+
+        /// <summary>
+        /// Gets or sets the affiliation property.
+        /// </summary>
+        /// <value>
+        ///  The owner.
```
Comment (M-Zuber) [inline Octokit/Models/Request/RepositoryRequest.cs]: I would think it should be `The affiliation`

**Comment by Sarmad93** [comment]
@M-Zuber is it right now? if there are further any mistakes please let me know. thanks

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 9]
```diff
@@ -103,4 +119,70 @@ public enum RepositorySort
         [Parameter(Value = "full_name")]
         FullName
     }
+
+
```
Comment (M-Zuber) [inline Octokit/Models/Request/RepositoryRequest.cs]: white space

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 10]
```diff
@@ -103,4 +119,70 @@ public enum RepositorySort
...
+        Private,
+
+        /// <summary>
+        /// Return both public and private repositories
+        /// </summary>
+        All,
+
+    }
+
+
```
Comment (M-Zuber) [inline Octokit/Models/Request/RepositoryRequest.cs]: white space

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 11]
```diff
@@ -103,4 +119,70 @@ public enum RepositorySort
...
+        /// </summary>
+        [Parameter(Value = "collaborator, organization_member")]
+        CollaboratorAndOrganizationMember,
+
+        /// <summary>
+        /// Returns all repositories where user is owner,collaborator or organization member.
+        /// </summary>
+        [Parameter(Value = "owner, collaborator, organization_member")]
+        All
+
```
Comment (M-Zuber) [inline Octokit/Models/Request/RepositoryRequest.cs]: white space

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 12]
```diff
@@ -37,6 +37,22 @@ public class RepositoryRequest : RequestParameters
...
+        /// <summary>
+        /// Gets or sets the visibility property.
+        /// </summary>
+        /// <value>
+        ///  The no.
+        /// </value>
+        public RepositoryVisibility Visibility { get; set; }
+
+        /// <summary>
+        /// Gets or sets The affiliation property.
```
Comment (M-Zuber) [inline Octokit/Models/Request/RepositoryRequest.cs]: `the` should probably be lower case

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 13]
```diff
@@ -37,6 +37,22 @@ public class RepositoryRequest : RequestParameters
...
+        /// </summary>
+        /// <value>
+        ///  The owner.
```
Comment (M-Zuber) [inline Octokit/Models/Request/RepositoryRequest.cs]: `The owner` should be changed to `The affiliation`

File: Octokit/Models/Request/RepositoryRequest.cs [hunk 14]
```diff
@@ -37,6 +37,22 @@ public class RepositoryRequest : RequestParameters
         /// </value>
         public SortDirection Direction { get; set; }
 
+        /// <summary>
+        /// Gets or sets the visibility property.
+        /// </summary>
+        /// <value>
+        ///  The no.
```
Comment (M-Zuber) [inline Octokit/Models/Request/RepositoryRequest.cs]: What do you mean by this?

**Comment by M-Zuber** [comment]
I apologize for being picky, it is an issue that I have.
Other then the little things I pointed out, it looks awesome :+1:

**Comment by Sarmad93** [comment]
I can understand your point as this is an official repository of github api these things(comment typos etc) really matter. I will push another commit when i will reach home.thanks again for correction. I appreciate it.

**Comment by M-Zuber** [comment]
:smile:

**Comment by Sarmad93** [comment]
Hi @M-Zuber  i need your help one more time :smile:  please review this when you are free.thanks

**Comment by M-Zuber** [comment]
LGTM, but I do not have commit rights, so you will have to wait for one of the maintainers to confirm and merge.
How do you feel about trying to add some tests for these changes in the meantime?

**Comment by ryangribble** [comment]
Regarding Unit Tests, I'd suggest it would be good to add a couple more similar to these:

   [```TheGetAllForCurrentMethod.CanFilterByType()```](https://github.com/octokit/octokit.net/blob/master/Octokit.Tests/Clients/RepositoriesClientTests.cs#L319)
   [```TheGetAllForCurrentMethod.CanFilterBySort()```](https://github.com/octokit/octokit.net/blob/master/Octokit.Tests/Clients/RepositoriesClientTests.cs#L338)
   [```TheGetAllForCurrentMethod.CanFilterBySortDirection()```](https://github.com/octokit/octokit.net/blob/master/Octokit.Tests/Clients/RepositoriesClientTests.cs#L359)

eg

CanFilterByVisibility
CanFilterByAffiliation

**Comment by Sarmad93** [comment]
@ryangribble unit tests :confused: it is difficult for me to write solid unit test i doubt that i will miss something in testing. it would be good if any one with testing knowledge should write test for these properties. what do you say?

**Comment by shiftkey** [comment]
@M-Zuber @ryangribble thanks for the initial review :gem:

@Sarmad93 I think writing these tests is actually easier that you suspect - we have a lot of tests already defined for `RepositoryClient`, and you can likely repurpose some of the existing tests that @ryangribble linked to.

The goal of these tests is to verify a small piece of behaviour - in this case, what we're interested in is the mapping of the parameters on the request to the URL that gets invoked by the client. The setup is what's called AAA syntax - Arrange, Act and Assert, and they're intended to be short, easy to read and quick to execute.

Have a read of the links above, feel free to ask questions - but don't be afraid to take a shot at it :grin:

**Comment by Sarmad93** [comment]
yes @shiftkey you are absolutely right :+1: I am new to this open source work that's why i am little afraid for now. As i was reading the links above i realized that i have lot to learn. Your code base has nice documentation and nice language idioms especially linq queries are written very well. what we learn in our programming classes is totally different here in open source world. Thanks :smile:

**Comment by Haacked** [comment]
> I am new to this open source work that's why i am little afraid for now.

Welcome!

We're all friendly here. We love having people help out in any way they see fit. We all have other responsibilities because of our work, so we can be slow to reply sometimes. But don't hesitate to ask questions. We want your first open source experience to be great so maybe you'll contribute again! :smile:

**Comment by Sarmad93** [comment]
Ok i will try to write unit test similar to above links tonight and will ask for the help if i get stuck.

**Comment by M-Zuber** [comment]
Good luck!

**Comment by Haacked** [comment]
Well done!

jgttfp

**Comment by M-Zuber** [comment]
Noice!! 
:shipit:

**Comment by Sarmad93** [comment]
Thank you everyone for helping me out especially @M-Zuber for initial review. You guys are really helpful.

**Comment by M-Zuber** [comment]
:blushing:

**Comment by Sarmad93** [comment]
@Haacked is it really you in the gif? :laughing:

**Comment by Haacked** [comment]
> @Haacked is it really you in the gif?

Yup!

**Pull request merged**